unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -254,6 +254,70 @@ public boolean equals(final Object other) {
     }
   }
 
+  /**
+   * Levenshtein distance is a metric for measuring the distance of two strings. The distance is
+   * defined by the minimum number of single-character edits (i.e. insertions, deletions or
+   * substitutions) that are required to change one of the strings into the other.
+   */
+  public int levenshteinDistance(UTF8String other) {
+    // Implementation adopted from org.apache.common.lang3.StringUtils.getLevenshteinDistance
+
+    int n = numChars();
```

Comment:
If each string will not exceed 65535 characters, you can use unsigned short maybe.